api/v2beta1/helmrelease_types.go

```diff
@@ -241,9 +241,9 @@ type Remediation interface {
 	MustIgnoreTestFailures(bool) bool
 	MustRemediateLastFailure() bool
 	GetStrategy() RemediationStrategy
-	GetFailureCount(hr HelmRelease) int64
+	GetFailureCount(hr *HelmRelease) int64
```

Comment:
I don't see much point in changing these from values to pointers; at least with values, it signals that the method does not mutate the value.

Reply:
(Since I can't put a comment on an unchanged line, it goes here: the Remediation interface should really be defined in the package that consumes arguments of that type, not here.)